IBAnimatable/Animatable.swift

```diff
@@ -66,44 +66,63 @@ public extension Animatable where Self: UIView {
     }
   }
 
-  public func animate(animation: AnimationType? = nil, completion: AnimatableCompletion? = nil) {
+  @discardableResult
+  public func animate(_ animation: AnimationType, duration: TimeInterval? = nil, damping: CGFloat? = nil, velocity: CGFloat? = nil, force: CGFloat? = nil) -> AnimationPromise<Self> {
+    return AnimationPromise(view: self).then(animation, duration: duration, damping: damping, velocity: velocity, force: force)
+  }
+
+  @available(*, deprecated, message: "use animate(AnimationType).then(...).completion instead. Will be removed in future versions")
+  public func animate(_ animation: AnimationType? = nil, completion: AnimatableCompletion? = nil) {// here for retro-compatibility
```

Comment:
Since we have this method, we don't have breaking API for `func animate(_ animation: AnimationType? = nil, completion: AnimatableCompletion? = nil)`. But shall we remove it for release 4?